BrainPortal/app/models/cluster_task.rb

```diff
@@ -485,13 +488,21 @@ def make_available(userfile, file_path, userfile_sub_path = nil, start_dir = nil
       File.unlink(full_path) if File.symlink?(full_path.to_s) # potential race condition here
     end
 
-    # Create the symlink
+
+    # Create the symlink or copy input data
     Dir.chdir(self.full_cluster_workdir) do
-      # Do nothing is symlink already exists with proper value.
-      # If there is something not a symlink in the way, or a symlink with a different
-      # value, the symlink() method will crash, which is what we want to
-      # catch the error in the situation.
+
+      # copying code
+      return userfile.cache_copy_to_local_file(full_path) if copy_file
+      # presently we do not save what (user)file is copied so no exception is raised
+      # if file or link exists already
+
+      # symlinking code
       unless File.symlink?(file_path.to_s) && File.readlink(file_path.to_s) == target.to_s
+        # Do nothing is symlink already exists with proper value.
```

Comment:
You moved the comment block that explains the logic of the `unless` statement UNDER the unless statement, why? That makes the code more difficult to understand.